tests/models/univnet/test_feature_extraction_univnet.py

```diff
@@ -0,0 +1,347 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+import os
+import random
+import tempfile
+import unittest
+
+import numpy as np
+from datasets import Audio, load_dataset
+
+from transformers import UnivNetFeatureExtractor
+from transformers.testing_utils import check_json_file_has_correct_format, require_torch, slow
+from transformers.utils.import_utils import is_torch_available
+
+from ...test_sequence_feature_extraction_common import SequenceFeatureExtractionTestMixin
+
+
+if is_torch_available():
+    import torch
+
+
+global_rng = random.Random()
+
+
```

Comment:
It turns out a bunch of models use this same function in their feature extraction tests, mostly without a `# Copied from` statement:

- **Whisper** ([tests/models/whisper/test_feature_extraction_whisper.py](https://github.com/huggingface/transformers/blob/main/tests/models/whisper/test_feature_extraction_whisper.py#L42))
- **TVLT** ([tests/models/tvlt/test_feature_extraction_tvlt.py](https://github.com/huggingface/transformers/blob/main/tests/models/tvlt/test_feature_extraction_tvlt.py#L41))
- **SpeechT5** ([tests/models/speecht5/test_feature_extraction_speecht5.py](https://github.com/huggingface/transformers/blob/main/tests/models/speecht5/test_feature_extraction_speecht5.py#L37))
- **Wav2Vec2** ([tests/models/wav2vec2/test_feature_extraction_wav2vec2.py](https://github.com/huggingface/transformers/blob/main/tests/models/wav2vec2/test_feature_extraction_wav2vec2.py#L32), [tests/models/wav2vec2/test_tokenization_wav2vec2.py](https://github.com/huggingface/transformers/blob/main/tests/models/wav2vec2/test_tokenization_wav2vec2.py#L41))
- **MusicGen** ([tests/models/musicgen/test_processing_musicgen.py](https://github.com/huggingface/transformers/blob/main/tests/models/musicgen/test_processing_musicgen.py#L38))
- **EnCodec** ([tests/models/encodec/test_feature_extraction_encodec.py](https://github.com/huggingface/transformers/blob/main/tests/models/encodec/test_feature_extraction_encodec.py#L37))
- **CLAP** ([tests/models/clap/test_feature_extraction_clap.py](https://github.com/huggingface/transformers/blob/main/tests/models/clap/test_feature_extraction_clap.py#L38))
- **Speech2Text** ([tests/models/speech_to_text/test_feature_extraction_speech_to_text.py](https://github.com/huggingface/transformers/blob/main/tests/models/speech_to_text/test_feature_extraction_speech_to_text.py#L35))
- **Audio Spectrogram Transformer** ([tests/models/audio_spectrogram_transformer/test_feature_extraction_audio_spectrogram_transformer.py](https://github.com/huggingface/transformers/blob/main/tests/models/audio_spectrogram_transformer/test_feature_extraction_audio_spectrogram_transformer.py#L36))

Since CLAP copies from the Whisper version, I will also copy from the Whisper version and consider it the "canonical" implementation. If you think it would be useful, I can open a PR to add `# Copied from` statements to the other models as well.

Reply:
Thanks for adding the `# Copied from` @dg845! Feel free as well to open a PR to add them to the rest of the models if you wish - otherwise let us know and one of us can :)

Reply:
I have opened a PR at #26581.